protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -619,37 +621,11 @@ func (rpccs *RPCConsumerServer) sendRelayToProvider(
 	}
 
 	result := make(chan *relayResponse)
-
-	go func(timeout time.Duration) {
-		responsesReceived := 0
-		relayReturned := false
-		for {
-			select {
-			case response := <-responses:
-				// increase responses received
-				responsesReceived++
-				if response.err == nil && !relayReturned {
-					// Return the first successful response
-					result <- response
-					relayReturned = true
-				}
-
-				if responsesReceived == len(sessions) {
-					// Return the last response if all previous responses were error
-					if !relayReturned {
-						result <- response
-					}
-
-					// if it was returned, just close this go routine
-					return
-				}
-			case <-time.After(relayTimeout + 2*time.Second):
-				// Timeout occurred, send an error to result channel
-				result <- &relayResponse{nil, NoResponseTimeout}
-				return
-			}
-		}
-	}(relayTimeout)
+	// Getting the best result from the providers,
+	// if there was an error we wait for the next result util timeout or a valid response
+	// priority order {valid response -> error response -> relay error}
+	// if there were multiple error responses picking the majority
+	go rpccs.getBestResult(result, relayTimeout, responses, len(sessions), chainMessage)
```

Comment:
this is a go routine with a blocking channel after, there's no need for that, just check the stuff in a sync function, and continue when it returns